spatial Voellmy friction model: add VariableVoellmyShapeToRaster to generate rasters from shapes [com6]

dwolfsch opened the pull request:
Adds functionality for generating raster files for mu and
xsi values from a DEM and shapefiles including:
A main script to rasterize mu and xsi based on polygon shapefiles and attribute fields.
A corresponding run script and configuration file for user customization


- [ ] as with the other PR add input setup for files -> @dwolfsch

dwolfsch committed:
spatial Voellmy friction model: add VariableVoellmyShapeToRaster to generate rasters from shapes

Adds functionality for generating raster files for mu and
xsi values from a DEM and shapefiles including:
A main script to rasterize mu and xsi based on polygon shapefiles and attribute fields.
A corresponding run script and configuration file for user customization.

pep8speaks commented:
Hello @dwolfsch! Thanks for opening this PR. 

* In the file [`avaframe/com6RockAvalanche/variableVoellmyShapeToRaster.py`](https://github.com/avaframe/AvaFrame/blob/d5624ec61b60ada195ca3f5b9b39af03a3329025/avaframe/com6RockAvalanche/variableVoellmyShapeToRaster.py):

> [Line 20:1](https://github.com/avaframe/AvaFrame/blob/d5624ec61b60ada195ca3f5b9b39af03a3329025/avaframe/com6RockAvalanche/variableVoellmyShapeToRaster.py#L20): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1

* In the file [`avaframe/in2Trans/shpConversion.py`](https://github.com/avaframe/AvaFrame/blob/d5624ec61b60ada195ca3f5b9b39af03a3329025/avaframe/in2Trans/shpConversion.py):

> [Line 431:52](https://github.com/avaframe/AvaFrame/blob/d5624ec61b60ada195ca3f5b9b39af03a3329025/avaframe/in2Trans/shpConversion.py#L431): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 429:48](https://github.com/avaframe/AvaFrame/blob/d5624ec61b60ada195ca3f5b9b39af03a3329025/avaframe/in2Trans/shpConversion.py#L429): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 428:48](https://github.com/avaframe/AvaFrame/blob/d5624ec61b60ada195ca3f5b9b39af03a3329025/avaframe/in2Trans/shpConversion.py#L428): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 143:35](https://github.com/avaframe/AvaFrame/blob/d5624ec61b60ada195ca3f5b9b39af03a3329025/avaframe/in2Trans/shpConversion.py#L143): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace

* In the file [`avaframe/runVariableVoellmyShapeToRaster.py`](https://github.com/avaframe/AvaFrame/blob/d5624ec61b60ada195ca3f5b9b39af03a3329025/avaframe/runVariableVoellmyShapeToRaster.py):

> [Line 54:1](https://github.com/avaframe/AvaFrame/blob/d5624ec61b60ada195ca3f5b9b39af03a3329025/avaframe/runVariableVoellmyShapeToRaster.py#L54): [E305](https://duckduckgo.com/?q=pep8%20E305) expected 2 blank lines after class or function definition, found 1
> [Line 15:1](https://github.com/avaframe/AvaFrame/blob/d5624ec61b60ada195ca3f5b9b39af03a3329025/avaframe/runVariableVoellmyShapeToRaster.py#L15): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1

Do see the [Hitchhiker's guide to code style](https://goo.gl/hqbW4r)

fso42 commented on avaframe/com6RockAvalanche/variableVoellmyShapeToRaster.ini:
```diff
@@ -0,0 +1,33 @@
+[INPUT]
+# Input DEM raster file path
+# Example: /path/to/dem.asc
+dem = 
+
+# Input shapefile for mu
+# Example: /path/to/mu_shapefile.shp
+# be aware, that the attribute name has to be "mu" 
+mu_shapefile = 
+
+# Input shapefile for xsi
+# Example: /path/to/xsi_shapefile.shp
+# be aware, that the attribute name has to be "xsi" 
+xsi_shapefile = 
```
These need to be setup as with the avalanche Dir info

fso42 commented on avaframe/runVariableVoellmyShapeToRaster.py:
```diff
@@ -0,0 +1,60 @@
+                        help='Path to the configuration file')
+
+    args = parser.parse_args()
+    runMuXsiWorkflow(str(args.configPath))
```
needs to be switched over to avaDir, (best to contact me for more explanation)

fso42 committed:
refactor(config): update default configuration values and comments

- Updated `variableVoellmyShapeToRasterCfg.ini` to use `mu` and `xsi` instead of Unicode characters for clarity.
- Modified `avaframeCfg.ini` to set a default path for `avalancheDir`.

fso42 committed:
refactor(spatialVoellmyInputs): replace and relocate `VariableVoellmyShapeToRaster` workflow

- Removed `variableVoellmyShapeToRaster` scripts and related configurations.
- Introduced `spatialVoellmyInputs` for handling spatial Voellmy friction parameter generation.
- Migrated default input values to `spatialVoellmyInputsCfg.ini` with updated defaults.
- Replaced inline raster generation logic with standardized utility functions.

fso42 committed:
docs(spatialVoellmyInputs): add documentation for generating rasters from shapefiles

awirb commented on avaframe/in3Utils/spatialVoellmyInputs.py:
```diff
@@ -0,0 +1,136 @@
+"""
+Functions for generating spatial Voellmy friction raster inputs.
```
could be a more general function to create rasters from shapefiles - consider renaming

fso42 replied:
At the moment it is hardcoded to mu xsi -> I'll stick with the naming

awirb commented on avaframe/in3Utils/spatialVoellmyInputs.py:
```diff
@@ -0,0 +1,136 @@
+    demShape = (demHeader["nrows"], demHeader["ncols"])
+
+    defaultMu = float(cfg["DEFAULTS"]["default_mu"])
+    defaultXsi = float(cfg["DEFAULTS"]["default_xsi"])
```
```suggestion
    defaultMu = cfg["DEFAULTS"].getfloat("default_mu")
    defaultXsi = float(cfg["DEFAULTS"].getfloat("default_xsi")
```

awirb commented on avaframe/in3Utils/spatialVoellmyInputs.py:
```diff
@@ -0,0 +1,136 @@
+
+
+def _rasterizeShapefile(shpPath, defaultValue, fieldName, demShape, demTransform):
+    """Rasterize a polygon shapefile attribute field onto a DEM-matching grid.
```
consider adding that all cells outside of polygons are filled with defaultValue

fso42 committed:
refactor(spatialVoellmyInputs): use `getfloat` for configuration parsing and improve docs

- Replaced `float` casting with `getfloat` for cleaner configuration parsing.
- Updated `_rasterizeShapefile` docstring to clarify default cell value behavior.
